### PR TITLE
vars list

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

document special vars

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

docs
##### ANSIBLE VERSION

```
2.7
```

### CODE REVIEW

File: docs/docsite/rst/user_guide/vars_list.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    The directory of the inventory source in which the `inventory_hostname` was first defined
+
+inventory_file
+    The file name of the inventory source in which the `inventory_hostname` was first defined
+
+omit
+    Special variable that allows you to 'omit' an option in a task, i.e ``- user: name=bob home={{ bobs_home|default(omit)}}``
+
+play_hosts
+    Deprecated, the same asn ansbile_play_batch
```
Comment (jamescassell) [inline docs/docsite/rst/user_guide/vars_list.rst]: asn -> as

File: docs/docsite/rst/user_guide/vars_list.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+playbook_dir
+    The path to the directory of the playbook that was passed to the ``ansible-playbook`` command line.
+
+role_names
+    The names of the rules currently imported into the current play.
+
+role_path
+    The path to the dir of the currently running role
+
```
Comment (jamescassell) [inline docs/docsite/rst/user_guide/vars_list.rst]: should `vars` show up here, since `hostvars` is above?

**Review by jamescassell — APPROVED** [review]
Awesome!  I've been looking for these docs for a while.  Glad to see they now exist!

shipit

File: docs/docsite/rst/user_guide/vars_list.rst (hunk 2 above)
Comment (bcoca) [reply docs/docsite/rst/user_guide/vars_list.rst]: no, vars is on my 'removal' list, its not documented for a reason

Comment (jamescassell) [reply docs/docsite/rst/user_guide/vars_list.rst]: Is there a proper way to get vars.keys() once vars is gone?

Comment (bcoca) [reply docs/docsite/rst/user_guide/vars_list.rst]: no, but that is a misleading usage, not all vars are in 'vars'

File: docs/docsite/rst/user_guide/vars_list.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+inventory_file
+    The file name of the inventory source in which the `inventory_hostname` was first defined
+
+omit
+    Special variable that allows you to 'omit' an option in a task, i.e ``- user: name=bob home={{ bobs_home|default(omit)}}``
+
+play_hosts
+    Deprecated, the same as ansbile_play_batch
+
+playbook_dir
```
Comment (jamescassell) [inline docs/docsite/rst/user_guide/vars_list.rst]: Would be great to have a `playbook_name` to get the filename of the running playbook.  Couldn't find the equivalent functionality

Comment (bcoca) [reply docs/docsite/rst/user_guide/vars_list.rst]: this pr is documenting existing feature, not for adding new ones, open a feature request for that

**Review by acozine — CHANGES_REQUESTED** [review]
This is fantastic content, thanks @bcoca for writing it up. I'd suggest a couple of changes:
1. Call the file `special_vars.rst`
2. Put it in the `reference_appendices` directory, since it's not task-focused
3. Add an anchor (`.. _special_variables:`)
4. Make the link from the `playbooks_variables` page a :ref: instead of a :doc:

**Review by acozine — APPROVED** [review]

**Pull request merged**